platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -0,0 +1,1330 @@
+	WindowData &wd = main_window;
+
+	if (wd.window_event_callback.is_valid()) {
+		Variant event = int(p_event);
+		wd.window_event_callback.call(event);
+	}
+}
+
+void DisplayServerWayland::dispatch_input_events(const Ref<InputEvent> &p_event) {
+	((DisplayServerWayland *)(get_singleton()))->_dispatch_input_event(p_event);
+}
+
+void DisplayServerWayland::_dispatch_input_event(const Ref<InputEvent> &p_event) {
+	Callable callable = main_window.input_event_callback;
+	if (callable.is_valid()) {
+		callable.call(p_event);
+	}
+}
+
+void DisplayServerWayland::_resize_window(const Size2i &p_size) {
+	WindowData &wd = main_window;
+
+	wd.rect.size = p_size;
+
+#ifdef VULKAN_ENABLED
+	if (wd.visible && context_vulkan) {
+		context_vulkan->window_resize(MAIN_WINDOW_ID, wd.rect.size.width, wd.rect.size.height);
+	}
+#endif
+
+#ifdef GLES3_ENABLED
+	if (wd.visible && egl_manager) {
+		wl_egl_window_resize(wd.wl_egl_window, wd.rect.size.width, wd.rect.size.height, 0, 0);
+	}
+#endif
+
+	if (wd.rect_changed_callback.is_valid()) {
+		wd.rect_changed_callback.call(wd.rect);
+	}
+}
+
+void DisplayServerWayland::_show_window() {
+	MutexLock mutex_lock(wayland_thread.mutex);
+
+	WindowData &wd = main_window;
+
+	if (!wd.visible) {
+		DEBUG_LOG_WAYLAND("Showing window.");
+
+		// Showing this window will reset its mode with whatever the compositor
+		// reports. We'll save the mode beforehand so that we can reapply it later.
+		// TODO: Fix/Port/Move/Whatever to `WaylandThread` APIs.
+		WindowMode setup_mode = wd.mode;
+
+		wayland_thread.window_create(MAIN_WINDOW_ID, wd.rect.size.width, wd.rect.size.height);
+		wayland_thread.window_set_min_size(MAIN_WINDOW_ID, wd.min_size);
+		wayland_thread.window_set_max_size(MAIN_WINDOW_ID, wd.max_size);
+		wayland_thread.window_set_app_id(MAIN_WINDOW_ID, _get_app_id_from_context(context));
+		wayland_thread.window_set_borderless(MAIN_WINDOW_ID, window_get_flag(WINDOW_FLAG_BORDERLESS));
+
+		// NOTE: The XDG shell protocol is built in a way that causes the window to
+		// be immediately shown as soon as a valid buffer is assigned to it. Hence,
+		// the only acceptable way of implementing window showing is to move the
+		// graphics context window creation logic here.
+#ifdef VULKAN_ENABLED
+		if (context_vulkan) {
+			struct wl_surface *wl_surface = wayland_thread.window_get_wl_surface(wd.id);
+			Error err = context_vulkan->window_create(MAIN_WINDOW_ID, wd.vsync_mode, wayland_thread.get_wl_display(), wl_surface, wd.rect.size.width, wd.rect.size.height);
+			ERR_FAIL_COND_MSG(err == ERR_CANT_CREATE, "Can't show a Vulkan window.");
+		}
+#endif
+
+#ifdef GLES3_ENABLED
+		if (egl_manager) {
+			struct wl_surface *wl_surface = wayland_thread.window_get_wl_surface(wd.id);
+			wd.wl_egl_window = wl_egl_window_create(wl_surface, wd.rect.size.width, wd.rect.size.height);
+
+			Error err = egl_manager->window_create(MAIN_WINDOW_ID, wayland_thread.get_wl_display(), wd.wl_egl_window, wd.rect.size.width, wd.rect.size.height);
```

Comment:
It seems `EGLManager::window_create` expects a `display*` and a `window*` as the names of the variables suggest:

https://github.com/godotengine/godot/blob/3ce73e5d419011d1642ed30a3417362b82c41898/drivers/egl/egl_manager.cpp#L200

Reply:
The manager doesn't send the values directly (for EGL != 1.5). It dereferences them before: 

https://github.com/godotengine/godot/blob/3ce73e5d419011d1642ed30a3417362b82c41898/drivers/egl/egl_manager.cpp#L61

https://github.com/godotengine/godot/blob/3ce73e5d419011d1642ed30a3417362b82c41898/drivers/egl/egl_manager.cpp#L219

Maybe that's not a bug related to your MR.